core/merginapi.cpp

```diff
@@ -2994,14 +2994,23 @@ bool MerginApi::hasLocalChanges(
   const QString &projectDir
 )
 {
-  if ( localFiles.count() != oldServerFiles.count() )
+  MerginConfig config = MerginConfig::fromFile( projectDir + "/" + sMerginConfigFile );
```

Comment:
There is a flag in `MerginApi` called `supportsSelectiveSync`. The code we add here should execute only if Mergin Api is set to use selective sync.